sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Customization/DatasetClient.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Analytics.Synapse.Artifacts
+{
+    public partial class DatasetClient
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DatasetClient"/>.
+        /// </summary>
+        public DatasetClient(Uri endpoint, TokenCredential credential)
```

Comment:
@tg-msft , thanks! This is a great suggestion and personally I think this is good for our customers so that they won't need to create multiple clients. However, during board review for Azure.Analytics.Synapse.Spark, we got some feedback: the reviewer suggests to remove the "top level" client because it doesn't have any API operations. It might be different scenarios for Azure.Analytics.Synapse.Spark and Azure.Analytics.Synapse.Artifacts because we only have 2 clients in Azure.Analytics.Synapse.Spark but we have have 11 clients in Azure.Analytics.Synapse.Artifacts.

What do you think?